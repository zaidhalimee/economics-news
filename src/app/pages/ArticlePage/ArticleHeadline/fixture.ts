const fixture = {
  type: 'headline',
  blocks: [
    {
      id: 'b209a5ab',
      type: 'text',
      model: {
        blocks: [
          {
            id: '1fe1e342',
            type: 'paragraph',
            model: {
              text: "'Inama ya gicuti' hagati ya Tshisekedi na Kagame - Qatar yabishoboye ite?",
              blocks: [
                {
                  id: 'bc738584',
                  type: 'fragment',
                  model: {
                    text: "'Inama ya gicuti' hagati ya Tshisekedi na Kagame - Qatar yabishoboye ite?",
                    attributes: [],
                  },
                  position: [1, 1, 1, 1],
                },
              ],
            },
            position: [1, 1, 1],
          },
        ],
      },
      position: [1, 1],
    },
  ],
};

export default fixture;
