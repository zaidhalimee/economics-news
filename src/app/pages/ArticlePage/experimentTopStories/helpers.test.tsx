import {
  getExperimentAnalyticsConfig,
  getExperimentTopStories,
} from './helpers';
import { topStoriesList } from '../PagePromoSections/TopStoriesSection/fixture/index';

jest.mock('../../../lib/analyticsUtils', () => ({
  ...jest.requireActual('../../../lib/analyticsUtils'),
  getAtUserId: jest.fn().mockReturnValue('123-456-789'),
}));

describe('AMP top stories experiment', () => {
  const mockTextBlock = {
    type: 'text',
    model: {
      blocks: [],
    },
  };
  const expectedExperimentTopStoriesBlock = (
    variant: 'Quarter' | 'Half' | 'ThreeQuarters',
  ) => {
    return {
      type: `experimentTopStories${variant}`,
      model: topStoriesList,
      id: `experimentTopStories${variant}`,
    };
  };

  const blocksEvenLength = Array(10).fill(mockTextBlock);
  const blocksOddLength = Array(11).fill(mockTextBlock);

  describe('getExperimentTopStories()', () => {
    it('returns shouldEnableExperimentTopStories as true if props match conditions.', () => {
      const { shouldEnableExperimentTopStories } = getExperimentTopStories({
        blocks: blocksEvenLength,
        topStoriesContent: topStoriesList,
        isAmp: true,
        id: 'c6v11qzyv8po',
        service: 'news',
      });
      expect(shouldEnableExperimentTopStories).toBe(true);
    });

    it.each`
      testDescription                                     | isAmp    | id                | service      | blocksLength
      ${'all props are undefined'}                        | ${false} | ${undefined}      | ${undefined} | ${undefined}
      ${'only isAmp is true'}                             | ${true}  | ${undefined}      | ${undefined} | ${undefined}
      ${'only id is undefined'}                           | ${true}  | ${undefined}      | ${'news'}    | ${11}
      ${'only id is defined and valid'}                   | ${false} | ${'c6v11qzyv8po'} | ${undefined} | ${undefined}
      ${'all props defined but id is invalid'}            | ${true}  | ${'c1231qzyv8po'} | ${'news'}    | ${11}
      ${'only service is undefined'}                      | ${true}  | ${'c6v11qzyv8po'} | ${undefined} | ${11}
      ${'only service is defined and valid'}              | ${false} | ${undefined}      | ${'news'}    | ${undefined}
      ${'all props defined but service is invalid'}       | ${true}  | ${'c6v11qzyv8po'} | ${'igbo'}    | ${11}
      ${'only blocks length is defined and valid'}        | ${false} | ${undefined}      | ${undefined} | ${11}
      ${'all props defined but blocks length is invalid'} | ${true}  | ${'c6v11qzyv8po'} | ${'news'}    | ${7}
    `(
      'returns shouldEnableExperimentTopStories as false because $testDescription.',
      ({ isAmp, id, service, blocksLength }) => {
        const blocks = Array(blocksLength).fill(mockTextBlock, 0);
        const { shouldEnableExperimentTopStories } = getExperimentTopStories({
          blocks,
          topStoriesContent: topStoriesList,
          isAmp,
          id,
          service,
        });

        expect(shouldEnableExperimentTopStories).toBe(false);
      },
    );

    const expectedBlocksEvenLength = [
      mockTextBlock,
      mockTextBlock,
      mockTextBlock,
      expectedExperimentTopStoriesBlock('Quarter'),
      mockTextBlock,
      expectedExperimentTopStoriesBlock('Half'),
      mockTextBlock,
      mockTextBlock,
      expectedExperimentTopStoriesBlock('ThreeQuarters'),
      mockTextBlock,
      mockTextBlock,
      mockTextBlock,
      mockTextBlock,
    ];
    const expectedBlocksOddLength = [
      mockTextBlock,
      mockTextBlock,
      mockTextBlock,
      expectedExperimentTopStoriesBlock('Quarter'),
      mockTextBlock,
      expectedExperimentTopStoriesBlock('Half'),
      mockTextBlock,
      mockTextBlock,
      mockTextBlock,
      expectedExperimentTopStoriesBlock('ThreeQuarters'),
      mockTextBlock,
      mockTextBlock,
      mockTextBlock,
      mockTextBlock,
    ];

    it.each`
      testType  | inputBlocks         | expectedOutput
      ${'even'} | ${blocksEvenLength} | ${expectedBlocksEvenLength}
      ${'odd'}  | ${blocksOddLength}  | ${expectedBlocksOddLength}
    `(
      'should insert experimentTopStories block into blocks array in the correct position when blocks.length is $testType.',
      ({ inputBlocks, expectedOutput }) => {
        const { transformedBlocks } = getExperimentTopStories({
          blocks: inputBlocks,
          topStoriesContent: topStoriesList,
          isAmp: true,
          id: 'c6v11qzyv8po',
          service: 'news',
        });
        expect(transformedBlocks).toEqual(expectedOutput);
      },
    );
  });

  describe('getExperimentAnalyticsConfig()', () => {
    process.env.SIMORGH_ATI_BASE_URL = 'http://foobar.com?';
    const NEWS_PRODUCER_ID = 64;
    const SPORT_PRODUCER_ID = 85;

    it.each`
      service    | env       | producerId
      ${'news'}  | ${'live'} | ${NEWS_PRODUCER_ID}
      ${'news'}  | ${'test'} | ${NEWS_PRODUCER_ID}
      ${'sport'} | ${'live'} | ${SPORT_PRODUCER_ID}
      ${'sport'} | ${'test'} | ${SPORT_PRODUCER_ID}
    `(
      'should create the analytics config with the correct parameters for $service on $env.',
      ({ env, service, producerId }) => {
        const analyticsConfig = getExperimentAnalyticsConfig({
          env,
          service,
          atiAnalyticsProducerId: producerId,
        });
        expect(analyticsConfig).toMatchSnapshot();
      },
    );
  });
});
