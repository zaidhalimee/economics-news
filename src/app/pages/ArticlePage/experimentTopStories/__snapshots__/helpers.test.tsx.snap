// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AMP top stories experiment getExperimentAnalyticsConfig() should create the analytics config with the correct parameters for news on live 1`] = `
{
  "requests": {
    "topStoriesArticleBodyClick": "http://foobar.com?idclient=123-456-789&s=$IF($EQUALS($MATCH(\${ampGeo}, gbOrUnknown, 0), gbOrUnknown), 598285, 598287)&s2=64&p=SOURCE_URL&r=\${screenWidth}x\${screenHeight}x\${screenColorDepth}&re=\${availableScreenWidth}x\${availableScreenHeight}&hl=\${timestamp}&lng=\${browserLanguage}&atc=PUB-[article]-[top-stories-promo]-[VARIANT(topStoriesExperiment)]-[]-[SOURCE_URL]-[articleBody]-[]-[]&mv_test=Google Discover&mv_experiment_id=topStoriesExperiment&mv_creation=VARIANT(topStoriesExperiment)&type=AT",
    "topStoriesArticleBodyView": "http://foobar.com?idclient=123-456-789&s=$IF($EQUALS($MATCH(\${ampGeo}, gbOrUnknown, 0), gbOrUnknown), 598285, 598287)&s2=64&p=SOURCE_URL&r=\${screenWidth}x\${screenHeight}x\${screenColorDepth}&re=\${availableScreenWidth}x\${availableScreenHeight}&hl=\${timestamp}&lng=\${browserLanguage}&ati=PUB-[article]-[top-stories-section]-[VARIANT(topStoriesExperiment)]-[]-[SOURCE_URL]-[articleBody]-[]-[]&mv_test=Google Discover&mv_experiment_id=topStoriesExperiment&mv_creation=VARIANT(topStoriesExperiment)&type=AT",
    "topStoriesSecondaryColumnClick": "http://foobar.com?idclient=123-456-789&s=$IF($EQUALS($MATCH(\${ampGeo}, gbOrUnknown, 0), gbOrUnknown), 598285, 598287)&s2=64&p=SOURCE_URL&r=\${screenWidth}x\${screenHeight}x\${screenColorDepth}&re=\${availableScreenWidth}x\${availableScreenHeight}&hl=\${timestamp}&lng=\${browserLanguage}&atc=PUB-[article]-[top-stories-promo]-[VARIANT(topStoriesExperiment)]-[]-[SOURCE_URL]-[secondaryColumn]-[]-[]&mv_test=Google Discover&mv_experiment_id=topStoriesExperiment&mv_creation=VARIANT(topStoriesExperiment)&type=AT",
    "topStoriesSecondaryColumnView": "http://foobar.com?idclient=123-456-789&s=$IF($EQUALS($MATCH(\${ampGeo}, gbOrUnknown, 0), gbOrUnknown), 598285, 598287)&s2=64&p=SOURCE_URL&r=\${screenWidth}x\${screenHeight}x\${screenColorDepth}&re=\${availableScreenWidth}x\${availableScreenHeight}&hl=\${timestamp}&lng=\${browserLanguage}&ati=PUB-[article]-[top-stories-section]-[VARIANT(topStoriesExperiment)]-[]-[SOURCE_URL]-[secondaryColumn]-[]-[]&mv_test=Google Discover&mv_experiment_id=topStoriesExperiment&mv_creation=VARIANT(topStoriesExperiment)&type=AT",
  },
  "triggers": {
    "articleBodyHalfPromoClick": {
      "on": "click",
      "request": "topStoriesArticleBodyClick",
      "selector": "div[data-experiment-position='articleBodyHalf'] a[aria-labelledby*='top-stories-promo']",
    },
    "articleBodyHalfView": {
      "on": "visible",
      "request": "topStoriesArticleBodyView",
      "visibilitySpec": {
        "continuousTimeMin": 200,
        "selector": "div[data-experiment-position='articleBodyHalf'] > section[aria-labelledby='top-stories-heading']",
        "totalTimeMin": 500,
        "visiblePercentageMin": 20,
      },
    },
    "articleBodyQuarterPromoClick": {
      "on": "click",
      "request": "topStoriesArticleBodyClick",
      "selector": "div[data-experiment-position='articleBodyQuarter'] a[aria-labelledby*='top-stories-promo']",
    },
    "articleBodyQuarterView": {
      "on": "visible",
      "request": "topStoriesArticleBodyView",
      "visibilitySpec": {
        "continuousTimeMin": 200,
        "selector": "div[data-experiment-position='articleBodyQuarter'] > section[aria-labelledby='top-stories-heading']",
        "totalTimeMin": 500,
        "visiblePercentageMin": 20,
      },
    },
    "articleBodyThreeQuartersPromoClick": {
      "on": "click",
      "request": "topStoriesArticleBodyClick",
      "selector": "div[data-experiment-position='articleBodyThreeQuarters'] a[aria-labelledby*='top-stories-promo']",
    },
    "articleBodyThreeQuartersView": {
      "on": "visible",
      "request": "topStoriesArticleBodyView",
      "visibilitySpec": {
        "continuousTimeMin": 200,
        "selector": "div[data-experiment-position='articleBodyThreeQuarters'] > section[aria-labelledby='top-stories-heading']",
        "totalTimeMin": 500,
        "visiblePercentageMin": 20,
      },
    },
    "secondaryColumnPromoClick": {
      "on": "click",
      "request": "topStoriesSecondaryColumnClick",
      "selector": "div[data-experiment-position='secondaryColumn'] a[aria-labelledby*='top-stories-promo']",
    },
    "secondaryColumnView": {
      "on": "visible",
      "request": "topStoriesSecondaryColumnView",
      "visibilitySpec": {
        "continuousTimeMin": 200,
        "selector": "div[data-experiment-position='secondaryColumn'] > section[aria-labelledby='top-stories-heading']",
        "totalTimeMin": 500,
        "visiblePercentageMin": 20,
      },
    },
  },
}
`;

exports[`AMP top stories experiment getExperimentAnalyticsConfig() should create the analytics config with the correct parameters for news on test 1`] = `
{
  "requests": {
    "topStoriesArticleBodyClick": "http://foobar.com?idclient=123-456-789&s=$IF($EQUALS($MATCH(\${ampGeo}, gbOrUnknown, 0), gbOrUnknown), 598286, 598288)&s2=64&p=SOURCE_URL&r=\${screenWidth}x\${screenHeight}x\${screenColorDepth}&re=\${availableScreenWidth}x\${availableScreenHeight}&hl=\${timestamp}&lng=\${browserLanguage}&atc=PUB-[article]-[top-stories-promo]-[VARIANT(topStoriesExperiment)]-[]-[SOURCE_URL]-[articleBody]-[]-[]&mv_test=Google Discover&mv_experiment_id=topStoriesExperiment&mv_creation=VARIANT(topStoriesExperiment)&type=AT",
    "topStoriesArticleBodyView": "http://foobar.com?idclient=123-456-789&s=$IF($EQUALS($MATCH(\${ampGeo}, gbOrUnknown, 0), gbOrUnknown), 598286, 598288)&s2=64&p=SOURCE_URL&r=\${screenWidth}x\${screenHeight}x\${screenColorDepth}&re=\${availableScreenWidth}x\${availableScreenHeight}&hl=\${timestamp}&lng=\${browserLanguage}&ati=PUB-[article]-[top-stories-section]-[VARIANT(topStoriesExperiment)]-[]-[SOURCE_URL]-[articleBody]-[]-[]&mv_test=Google Discover&mv_experiment_id=topStoriesExperiment&mv_creation=VARIANT(topStoriesExperiment)&type=AT",
    "topStoriesSecondaryColumnClick": "http://foobar.com?idclient=123-456-789&s=$IF($EQUALS($MATCH(\${ampGeo}, gbOrUnknown, 0), gbOrUnknown), 598286, 598288)&s2=64&p=SOURCE_URL&r=\${screenWidth}x\${screenHeight}x\${screenColorDepth}&re=\${availableScreenWidth}x\${availableScreenHeight}&hl=\${timestamp}&lng=\${browserLanguage}&atc=PUB-[article]-[top-stories-promo]-[VARIANT(topStoriesExperiment)]-[]-[SOURCE_URL]-[secondaryColumn]-[]-[]&mv_test=Google Discover&mv_experiment_id=topStoriesExperiment&mv_creation=VARIANT(topStoriesExperiment)&type=AT",
    "topStoriesSecondaryColumnView": "http://foobar.com?idclient=123-456-789&s=$IF($EQUALS($MATCH(\${ampGeo}, gbOrUnknown, 0), gbOrUnknown), 598286, 598288)&s2=64&p=SOURCE_URL&r=\${screenWidth}x\${screenHeight}x\${screenColorDepth}&re=\${availableScreenWidth}x\${availableScreenHeight}&hl=\${timestamp}&lng=\${browserLanguage}&ati=PUB-[article]-[top-stories-section]-[VARIANT(topStoriesExperiment)]-[]-[SOURCE_URL]-[secondaryColumn]-[]-[]&mv_test=Google Discover&mv_experiment_id=topStoriesExperiment&mv_creation=VARIANT(topStoriesExperiment)&type=AT",
  },
  "triggers": {
    "articleBodyHalfPromoClick": {
      "on": "click",
      "request": "topStoriesArticleBodyClick",
      "selector": "div[data-experiment-position='articleBodyHalf'] a[aria-labelledby*='top-stories-promo']",
    },
    "articleBodyHalfView": {
      "on": "visible",
      "request": "topStoriesArticleBodyView",
      "visibilitySpec": {
        "continuousTimeMin": 200,
        "selector": "div[data-experiment-position='articleBodyHalf'] > section[aria-labelledby='top-stories-heading']",
        "totalTimeMin": 500,
        "visiblePercentageMin": 20,
      },
    },
    "articleBodyQuarterPromoClick": {
      "on": "click",
      "request": "topStoriesArticleBodyClick",
      "selector": "div[data-experiment-position='articleBodyQuarter'] a[aria-labelledby*='top-stories-promo']",
    },
    "articleBodyQuarterView": {
      "on": "visible",
      "request": "topStoriesArticleBodyView",
      "visibilitySpec": {
        "continuousTimeMin": 200,
        "selector": "div[data-experiment-position='articleBodyQuarter'] > section[aria-labelledby='top-stories-heading']",
        "totalTimeMin": 500,
        "visiblePercentageMin": 20,
      },
    },
    "articleBodyThreeQuartersPromoClick": {
      "on": "click",
      "request": "topStoriesArticleBodyClick",
      "selector": "div[data-experiment-position='articleBodyThreeQuarters'] a[aria-labelledby*='top-stories-promo']",
    },
    "articleBodyThreeQuartersView": {
      "on": "visible",
      "request": "topStoriesArticleBodyView",
      "visibilitySpec": {
        "continuousTimeMin": 200,
        "selector": "div[data-experiment-position='articleBodyThreeQuarters'] > section[aria-labelledby='top-stories-heading']",
        "totalTimeMin": 500,
        "visiblePercentageMin": 20,
      },
    },
    "secondaryColumnPromoClick": {
      "on": "click",
      "request": "topStoriesSecondaryColumnClick",
      "selector": "div[data-experiment-position='secondaryColumn'] a[aria-labelledby*='top-stories-promo']",
    },
    "secondaryColumnView": {
      "on": "visible",
      "request": "topStoriesSecondaryColumnView",
      "visibilitySpec": {
        "continuousTimeMin": 200,
        "selector": "div[data-experiment-position='secondaryColumn'] > section[aria-labelledby='top-stories-heading']",
        "totalTimeMin": 500,
        "visiblePercentageMin": 20,
      },
    },
  },
}
`;

exports[`AMP top stories experiment getExperimentAnalyticsConfig() should create the analytics config with the correct parameters for sport on live 1`] = `
{
  "requests": {
    "topStoriesArticleBodyClick": "http://foobar.com?idclient=123-456-789&s=$IF($EQUALS($MATCH(\${ampGeo}, gbOrUnknown, 0), gbOrUnknown), 598310, 598308)&s2=85&p=SOURCE_URL&r=\${screenWidth}x\${screenHeight}x\${screenColorDepth}&re=\${availableScreenWidth}x\${availableScreenHeight}&hl=\${timestamp}&lng=\${browserLanguage}&atc=PUB-[article]-[top-stories-promo]-[VARIANT(topStoriesExperiment)]-[]-[SOURCE_URL]-[articleBody]-[]-[]&mv_test=Google Discover&mv_experiment_id=topStoriesExperiment&mv_creation=VARIANT(topStoriesExperiment)&type=AT",
    "topStoriesArticleBodyView": "http://foobar.com?idclient=123-456-789&s=$IF($EQUALS($MATCH(\${ampGeo}, gbOrUnknown, 0), gbOrUnknown), 598310, 598308)&s2=85&p=SOURCE_URL&r=\${screenWidth}x\${screenHeight}x\${screenColorDepth}&re=\${availableScreenWidth}x\${availableScreenHeight}&hl=\${timestamp}&lng=\${browserLanguage}&ati=PUB-[article]-[top-stories-section]-[VARIANT(topStoriesExperiment)]-[]-[SOURCE_URL]-[articleBody]-[]-[]&mv_test=Google Discover&mv_experiment_id=topStoriesExperiment&mv_creation=VARIANT(topStoriesExperiment)&type=AT",
    "topStoriesSecondaryColumnClick": "http://foobar.com?idclient=123-456-789&s=$IF($EQUALS($MATCH(\${ampGeo}, gbOrUnknown, 0), gbOrUnknown), 598310, 598308)&s2=85&p=SOURCE_URL&r=\${screenWidth}x\${screenHeight}x\${screenColorDepth}&re=\${availableScreenWidth}x\${availableScreenHeight}&hl=\${timestamp}&lng=\${browserLanguage}&atc=PUB-[article]-[top-stories-promo]-[VARIANT(topStoriesExperiment)]-[]-[SOURCE_URL]-[secondaryColumn]-[]-[]&mv_test=Google Discover&mv_experiment_id=topStoriesExperiment&mv_creation=VARIANT(topStoriesExperiment)&type=AT",
    "topStoriesSecondaryColumnView": "http://foobar.com?idclient=123-456-789&s=$IF($EQUALS($MATCH(\${ampGeo}, gbOrUnknown, 0), gbOrUnknown), 598310, 598308)&s2=85&p=SOURCE_URL&r=\${screenWidth}x\${screenHeight}x\${screenColorDepth}&re=\${availableScreenWidth}x\${availableScreenHeight}&hl=\${timestamp}&lng=\${browserLanguage}&ati=PUB-[article]-[top-stories-section]-[VARIANT(topStoriesExperiment)]-[]-[SOURCE_URL]-[secondaryColumn]-[]-[]&mv_test=Google Discover&mv_experiment_id=topStoriesExperiment&mv_creation=VARIANT(topStoriesExperiment)&type=AT",
  },
  "triggers": {
    "articleBodyHalfPromoClick": {
      "on": "click",
      "request": "topStoriesArticleBodyClick",
      "selector": "div[data-experiment-position='articleBodyHalf'] a[aria-labelledby*='top-stories-promo']",
    },
    "articleBodyHalfView": {
      "on": "visible",
      "request": "topStoriesArticleBodyView",
      "visibilitySpec": {
        "continuousTimeMin": 200,
        "selector": "div[data-experiment-position='articleBodyHalf'] > section[aria-labelledby='top-stories-heading']",
        "totalTimeMin": 500,
        "visiblePercentageMin": 20,
      },
    },
    "articleBodyQuarterPromoClick": {
      "on": "click",
      "request": "topStoriesArticleBodyClick",
      "selector": "div[data-experiment-position='articleBodyQuarter'] a[aria-labelledby*='top-stories-promo']",
    },
    "articleBodyQuarterView": {
      "on": "visible",
      "request": "topStoriesArticleBodyView",
      "visibilitySpec": {
        "continuousTimeMin": 200,
        "selector": "div[data-experiment-position='articleBodyQuarter'] > section[aria-labelledby='top-stories-heading']",
        "totalTimeMin": 500,
        "visiblePercentageMin": 20,
      },
    },
    "articleBodyThreeQuartersPromoClick": {
      "on": "click",
      "request": "topStoriesArticleBodyClick",
      "selector": "div[data-experiment-position='articleBodyThreeQuarters'] a[aria-labelledby*='top-stories-promo']",
    },
    "articleBodyThreeQuartersView": {
      "on": "visible",
      "request": "topStoriesArticleBodyView",
      "visibilitySpec": {
        "continuousTimeMin": 200,
        "selector": "div[data-experiment-position='articleBodyThreeQuarters'] > section[aria-labelledby='top-stories-heading']",
        "totalTimeMin": 500,
        "visiblePercentageMin": 20,
      },
    },
    "secondaryColumnPromoClick": {
      "on": "click",
      "request": "topStoriesSecondaryColumnClick",
      "selector": "div[data-experiment-position='secondaryColumn'] a[aria-labelledby*='top-stories-promo']",
    },
    "secondaryColumnView": {
      "on": "visible",
      "request": "topStoriesSecondaryColumnView",
      "visibilitySpec": {
        "continuousTimeMin": 200,
        "selector": "div[data-experiment-position='secondaryColumn'] > section[aria-labelledby='top-stories-heading']",
        "totalTimeMin": 500,
        "visiblePercentageMin": 20,
      },
    },
  },
}
`;

exports[`AMP top stories experiment getExperimentAnalyticsConfig() should create the analytics config with the correct parameters for sport on test 1`] = `
{
  "requests": {
    "topStoriesArticleBodyClick": "http://foobar.com?idclient=123-456-789&s=$IF($EQUALS($MATCH(\${ampGeo}, gbOrUnknown, 0), gbOrUnknown), 598311, 598309)&s2=85&p=SOURCE_URL&r=\${screenWidth}x\${screenHeight}x\${screenColorDepth}&re=\${availableScreenWidth}x\${availableScreenHeight}&hl=\${timestamp}&lng=\${browserLanguage}&atc=PUB-[article]-[top-stories-promo]-[VARIANT(topStoriesExperiment)]-[]-[SOURCE_URL]-[articleBody]-[]-[]&mv_test=Google Discover&mv_experiment_id=topStoriesExperiment&mv_creation=VARIANT(topStoriesExperiment)&type=AT",
    "topStoriesArticleBodyView": "http://foobar.com?idclient=123-456-789&s=$IF($EQUALS($MATCH(\${ampGeo}, gbOrUnknown, 0), gbOrUnknown), 598311, 598309)&s2=85&p=SOURCE_URL&r=\${screenWidth}x\${screenHeight}x\${screenColorDepth}&re=\${availableScreenWidth}x\${availableScreenHeight}&hl=\${timestamp}&lng=\${browserLanguage}&ati=PUB-[article]-[top-stories-section]-[VARIANT(topStoriesExperiment)]-[]-[SOURCE_URL]-[articleBody]-[]-[]&mv_test=Google Discover&mv_experiment_id=topStoriesExperiment&mv_creation=VARIANT(topStoriesExperiment)&type=AT",
    "topStoriesSecondaryColumnClick": "http://foobar.com?idclient=123-456-789&s=$IF($EQUALS($MATCH(\${ampGeo}, gbOrUnknown, 0), gbOrUnknown), 598311, 598309)&s2=85&p=SOURCE_URL&r=\${screenWidth}x\${screenHeight}x\${screenColorDepth}&re=\${availableScreenWidth}x\${availableScreenHeight}&hl=\${timestamp}&lng=\${browserLanguage}&atc=PUB-[article]-[top-stories-promo]-[VARIANT(topStoriesExperiment)]-[]-[SOURCE_URL]-[secondaryColumn]-[]-[]&mv_test=Google Discover&mv_experiment_id=topStoriesExperiment&mv_creation=VARIANT(topStoriesExperiment)&type=AT",
    "topStoriesSecondaryColumnView": "http://foobar.com?idclient=123-456-789&s=$IF($EQUALS($MATCH(\${ampGeo}, gbOrUnknown, 0), gbOrUnknown), 598311, 598309)&s2=85&p=SOURCE_URL&r=\${screenWidth}x\${screenHeight}x\${screenColorDepth}&re=\${availableScreenWidth}x\${availableScreenHeight}&hl=\${timestamp}&lng=\${browserLanguage}&ati=PUB-[article]-[top-stories-section]-[VARIANT(topStoriesExperiment)]-[]-[SOURCE_URL]-[secondaryColumn]-[]-[]&mv_test=Google Discover&mv_experiment_id=topStoriesExperiment&mv_creation=VARIANT(topStoriesExperiment)&type=AT",
  },
  "triggers": {
    "articleBodyHalfPromoClick": {
      "on": "click",
      "request": "topStoriesArticleBodyClick",
      "selector": "div[data-experiment-position='articleBodyHalf'] a[aria-labelledby*='top-stories-promo']",
    },
    "articleBodyHalfView": {
      "on": "visible",
      "request": "topStoriesArticleBodyView",
      "visibilitySpec": {
        "continuousTimeMin": 200,
        "selector": "div[data-experiment-position='articleBodyHalf'] > section[aria-labelledby='top-stories-heading']",
        "totalTimeMin": 500,
        "visiblePercentageMin": 20,
      },
    },
    "articleBodyQuarterPromoClick": {
      "on": "click",
      "request": "topStoriesArticleBodyClick",
      "selector": "div[data-experiment-position='articleBodyQuarter'] a[aria-labelledby*='top-stories-promo']",
    },
    "articleBodyQuarterView": {
      "on": "visible",
      "request": "topStoriesArticleBodyView",
      "visibilitySpec": {
        "continuousTimeMin": 200,
        "selector": "div[data-experiment-position='articleBodyQuarter'] > section[aria-labelledby='top-stories-heading']",
        "totalTimeMin": 500,
        "visiblePercentageMin": 20,
      },
    },
    "articleBodyThreeQuartersPromoClick": {
      "on": "click",
      "request": "topStoriesArticleBodyClick",
      "selector": "div[data-experiment-position='articleBodyThreeQuarters'] a[aria-labelledby*='top-stories-promo']",
    },
    "articleBodyThreeQuartersView": {
      "on": "visible",
      "request": "topStoriesArticleBodyView",
      "visibilitySpec": {
        "continuousTimeMin": 200,
        "selector": "div[data-experiment-position='articleBodyThreeQuarters'] > section[aria-labelledby='top-stories-heading']",
        "totalTimeMin": 500,
        "visiblePercentageMin": 20,
      },
    },
    "secondaryColumnPromoClick": {
      "on": "click",
      "request": "topStoriesSecondaryColumnClick",
      "selector": "div[data-experiment-position='secondaryColumn'] a[aria-labelledby*='top-stories-promo']",
    },
    "secondaryColumnView": {
      "on": "visible",
      "request": "topStoriesSecondaryColumnView",
      "visibilitySpec": {
        "continuousTimeMin": 200,
        "selector": "div[data-experiment-position='secondaryColumn'] > section[aria-labelledby='top-stories-heading']",
        "totalTimeMin": 500,
        "visiblePercentageMin": 20,
      },
    },
  },
}
`;
